packages/jest-jasmine2/src/reporter.js

```diff
@@ -140,7 +140,7 @@ class Jasmine2Reporter {
     };
 
     specResult.failedExpectations.forEach(failed => {
-      const message = (!failed.matcherName && failed.stack)
+      const message = !failed.matcherName && failed.stack
```

Comment:
yeah, it makes sense here because it isn't ambiguous. They are only added if && or || are used.